src/adaptation/icap/Xaction.cc

```diff
@@ -442,6 +442,7 @@ void Adaptation::Icap::Xaction::noteCommRead(const CommIoCbParams &io)
 
     CommIoCbParams rd(this); // will be expanded with ReadNow results
     rd.conn = io.conn;
+    rd.size = SQUID_TCP_SO_RCVBUF - readBuf.length();
```

Comment:
It feels like we need to fix this case too, although I could not make readBuf.length() grow more than SQUID_TCP_SO_RCVBUF  in my tests.